action/protocol/execution/evm/evmstatedbadapter.go

```diff
@@ -318,11 +342,14 @@ func (stateDB *StateDBAdapter) SetNonce(evmAddr common.Address, nonce uint64) {
 	log.L().Debug("Called SetNonce.",
 		zap.String("address", addr.String()),
 		zap.Uint64("nonce", nonce))
-	s.Nonce = nonce
-	if err := accountutil.StoreAccount(stateDB.sm, addr, s); err != nil {
+	if err := accountutil.SetNonce(s, nonce); err != nil {
```

Comment:
i don't recall, in the case `stateDB.usePendingNonce == true`, why in `GetNonce()` need to do `nonce++` and in `SetNonce()` do `nonce--` to bring it back (line 340 above)

does this logic still hold for the new account (type = 1)?